packages/block-editor/src/components/inner-blocks/index.js

```diff
@@ -139,12 +139,13 @@ InnerBlocks = compose( [
 			getTemplateLock,
 		} = select( 'core/block-editor' );
 		const { clientId } = ownProps;
+		const block = getBlock( clientId );
 		const rootClientId = getBlockRootClientId( clientId );
 
 		return {
-			block: getBlock( clientId ),
+			block,
 			blockListSettings: getBlockListSettings( clientId ),
-			hasOverlay: ! isBlockSelected( clientId ) && ! hasSelectedInnerBlock( clientId, true ),
+			hasOverlay: block.name !== 'core/template' && ! isBlockSelected( clientId ) && ! hasSelectedInnerBlock( clientId, true ),
```

Comment:
@youknowriad - any thoughts on this one? It's a really nasty hack to avoid having the overlay in the context of nesting...

Reply:
I think it's fine, ideally, this `core/template` block should be removed. Rebuilding the reusable blocks UI as a nested editor would allow us to remove it. See #14367

That PR is blocked by the fact that it's hard to use nested Slot/Fills. One tradeoff we can make to unblock it could be to decide that we don't show the inspector settings for the reusable blocks content and redirect the user to the edit page of the reusable blocks for a full editing session.